### PR TITLE
adaptive simulcast 

### PR DESCRIPTION
**Pull request by makarandp0**
This PR merges adaptive simulcast changes into develop. We will cut an RC from develop.
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,42 @@ The Twilio Programmable Video SDKs use [Semantic Versioning](http://www.semver.o
 
 **Version 1.x reached End of Life on September 8th, 2021.** See the changelog entry [here](https://www.twilio.com/changelog/end-of-life-complete-for-unsupported-versions-of-the-programmable-video-sdk). Support for the 1.x version ended on December 4th, 2020.
 
+2.19.0 (In Progress)
+====================
+New Features
+------------
+
+This release introduces a new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in ConnectOptions. When joining a group room with this feature enabled, the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side. The beta currently does not support setting a max bitrate for your simulcast layers.
```
Comment (manjeshbhargav) [inline CHANGELOG.md:10]: Please make this multiline, and add a bullet point.
```suggestion
- This release introduces a new beta feature **Adaptive Simulcast**. This opt-in feature can be enabled by setting `preferredVideoCodecs="auto"` in ConnectOptions. When joining a group room with this feature enabled, the SDK will use VP8 simulcast, and will enable/disable simulcast layers dynamically, thus improving bandwidth and CPU usage. It works best when used along with `Client Track Switch Off Controls` and `Video Content Preferences`. These two flags allows the SFU to determine which simulcast layers are needed, thus allowing it to disable the layers not needed on publisher side. The beta currently does not support setting a max bitrate for your simulcast layers.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,42 @@ The Twilio Programmable Video SDKs use [Semantic Versioning](http://www.semver.o
 
 **Version 1.x reached End of Life on September 8th, 2021.** See the changelog entry [here](https://www.twilio.com/changelog/end-of-life-complete-for-unsupported-versions-of-the-programmable-video-sdk). Support for the 1.x version ended on December 4th, 2020.
 
+2.19.0 (In Progress)
+====================
```
Comment (manjeshbhargav) [inline CHANGELOG.md:6]:
```suggestion
====================

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,42 @@ The Twilio Programmable Video SDKs use [Semantic Versioning](http://www.semver.o
...
+  const room = await connect(token, {
+    preferredVideoCodecs: 'auto',
+    bandwidthProfile: {
+      video: {
+        contentPreferencesMode: 'auto',
+        clientTrackSwitchOffControl: 'auto'
+      }
+    }
+  });
+  ```
```
Comment (manjeshbhargav) [inline CHANGELOG.md:28]:
```suggestion
Example:

```ts
const { connect } = require('twilio-video');

const room = await connect(token, {
  preferredVideoCodecs: 'auto',
  bandwidthProfile: {
    video: {
      contentPreferencesMode: 'auto',
      clientTrackSwitchOffControl: 'auto'
    }
  }
});
\`\`\`
```

**Comment by makarandp0** [comment]
Unit tests are failing for unrelated reason. I will merge this change now, and work on fixing unit tests in `develop` branch.

**Pull request merged**